src/zenml/pipelines/base_pipeline.py

```diff
@@ -1018,3 +976,101 @@ def _get_latest_version(self) -> Optional[int]:
             return int(all_pipelines.items[0].version)
         else:
             return None
+
+    def _load_or_create_pipeline_build(
+        self,
+        deployment: "PipelineDeployment",
+        pipeline_spec: "PipelineSpec",
+        build: Union[str, "UUID", "PipelineBuild", None] = None,
+    ) -> Optional["PipelineBuildResponseModel"]:
+        build_model = None
+
+        if build:
+            logger.info(
+                "Using an old build for a pipeline run can lead to "
+                "unexpected behavior as the pipeline will run with the step "
+                "code that was included in the Docker images which might "
+                "differ from the code in your client environment."
+            )
+
+            if isinstance(build, str):
+                build = UUID(build, version=4)
+
+            if isinstance(build, UUID):
+                build_model = Client().zen_store.get_build(build_id=build)
+
+                if build_model.pipeline:
+                    build_hash = build_model.pipeline.version_hash
+                    current_hash = self._compute_unique_identifier(
+                        pipeline_spec=pipeline_spec
+                    )
+
+                    if build_hash != current_hash:
+                        logger.warning(
+                            "The pipeline associated with the build you "
+                            "specified for this run has a different spec "
+                            "or step code. This might lead to unexpected "
+                            "behavior as this pipeline run will use the "
+                            "code that was included in the Docker images which "
+                            "might differ from the code in your client "
+                            "environment."
+                        )
+            else:
```

Comment:
```suggestion
            elif not isinstance(build, PipelineBuild):
                raise ....
            else:
```

Reply:
At this point it can only be a `PipelineBuild` right? It already checked for None, str and UUID